### PR TITLE
Consider ReferenceGrant in HTTPRoute Service validation KIA1402

### PR DESCRIPTION
**Pull request by hhovsepy**
Issue https://github.com/kiali/kiali/issues/6918

In K8s HTTPRoute Backend reference (https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.BackendObjectReference), when the Namespace where Service is referenced differs from the local namespace, where the HTTPRoute is created, then a ReferenceGrant object is required to reference the Service namespace and local namespace.

Now: KIA1402 validation is modified to consider ReferenceGrant objects when those 2 namespaces are different. 

How to test it:
Apply K8s Gateway API CRD v1.0.0
Create HTTPRoute and ReferenceGrant.
```
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: inbookinfo
  namespace: bookinfo
spec:
  rules:
  - matches:
    - path:
        type: Exact
        value: "/bar"
    backendRefs:
      - name: kiali
        namespace: istio-system
        port: 9080
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: ReferenceGrant
metadata:
  name: toistio
  namespace: istio-system
spec:
  from:
  - group: gateway.networking.k8s.io
    kind: HTTPRoute
    namespace: bookinfo
  to:
  - group: ""
    kind: Service
```
Verify that HTTPRoute is referring to a Service from remote namespace, but no error as ReferenceGrant is referencing those namespaces.
Screenshot from 2023-12-20 19-54-30
Screenshot from 2023-12-20 19-52-32

When ReferenceGrant does not exist:
Screenshot from 2023-12-20 19-54-51

### CODE REVIEW

File: business/checkers/k8shttproute_checker.go [hunk 1]
```diff
@@ -11,11 +12,12 @@ import (
 const K8sHTTPRouteCheckerType = "k8shttproute"
 
 type K8sHTTPRouteChecker struct {
-	K8sHTTPRoutes    []*k8s_networking_v1.HTTPRoute
-	K8sGateways      []*k8s_networking_v1.Gateway
-	Namespaces       models.Namespaces
-	RegistryServices []*kubernetes.RegistryService
-	Cluster          string
+	K8sHTTPRoutes      []*k8s_networking_v1.HTTPRoute
```
Comment (josunect) [inline business/checkers/k8shttproute_checker.go:15]: Just a minor thing, it would be possible to alphabetize?

**Review by josunect — COMMENTED** [review]
Verified: 
image

With the reference grant: 
image

Just a minor comment.

File: business/checkers/k8shttproute_checker.go (hunk 1 above)
Comment (hhovsepy) [reply business/checkers/k8shttproute_checker.go:15]: ah right

**Review by josunect — APPROVED** [review]
LGTM 

Thanks, @hhovsepy !

**Pull request merged**